### PR TITLE
[ALLUXIO-3156] Use unique ID as domain socket address

### PR DESCRIPTION
**Pull request by madanadit**
JIRA: https://alluxio.atlassian.net/browse/ALLUXIO-3156

This PR introduces a new property `alluxio.worker.data.server.domain.socket.as.worker.id` to enable using the worker id as domain socket address. 

In k8s, where a compute framework may run with virtual networking, the client hostname may not match worker hostname. In such a scenario, this new property is used to enable short circuit by inspecting the filesystem instead of comparing the network address.

### CODE REVIEW

**Comment by madanadit** [comment]
alluxio-bot, check this please

**Comment by madanadit** [comment]
@calvinjia @aaudiber I changed the logic to use a uuid generated locally at the worker instead of the master getWorkerId rpc. Please take a look. I'm adding docs now

File: core/common/src/main/java/alluxio/util/network/NettyUtils.java [hunk 1]
```diff
@@ -149,9 +150,16 @@ public static void disableAutoRead(Channel channel) {
    * @return true if the domain socket is enabled on this client
    */
   public static boolean isDomainSocketSupported(WorkerNetAddress workerNetAddress) {
-    return workerNetAddress.getHost().equals(NetworkAddressUtils.getClientHostName())
-        && !workerNetAddress.getDomainSocketPath().isEmpty()
-        && CHANNEL_TYPE == ChannelType.EPOLL;
+    boolean isDomainSocketSupported = !workerNetAddress.getDomainSocketPath().isEmpty();
```
Comment (aaudiber) [inline core/common/src/main/java/alluxio/util/network/NettyUtils.java]: we could simplify this by immediately returning `false` if `workerNetAddress.getDomainSocketPath().isEmpty()`

File: core/server/worker/src/main/java/alluxio/worker/AlluxioWorkerProcess.java [hunk 2]
```diff
@@ -151,9 +153,14 @@ public Void call() throws Exception {
       mDataServer = DataServer.Factory
           .create(NetworkAddressUtils.getBindAddress(ServiceType.WORKER_DATA), this);
 
+      // Setup domain socket data server
       if (isDomainSocketEnabled()) {
         String domainSocketPath =
             Configuration.get(PropertyKey.WORKER_DATA_SERVER_DOMAIN_SOCKET_ADDRESS);
+        if (Configuration.getBoolean(PropertyKey.WORKER_DATA_SERVER_DOMAIN_SOCKET_AS_UUID)) {
+          domainSocketPath =
+              PathUtils.concatPath(domainSocketPath, IdUtils.getRandomNonNegativeLong());
```
Comment (aaudiber) [inline core/server/worker/src/main/java/alluxio/worker/AlluxioWorkerProcess.java]: it's weird that we call this a UUID, but it's actually a random non-negative long

**Comment by madanadit** [comment]
@aaudiber Thanks. Addressed comments

**Comment by madanadit** [comment]
Added a short section to the k8s page in this PR. The kubernetes test will test the new functionality. I looked into adding integration tests but that seems hard because of domain socket on mac.

**Comment by calvinjia** [comment]
@madanadit Is there any case where the UUID approach would not be preferred?

**Comment by madanadit** [comment]
I think UUID is the preferred approach going forward. I decided to introduce a property w/ false as default for backward compatibility.

**Pull request merged**